platforms/core-configuration/model-core/src/main/java/org/gradle/api/internal/provider/AbstractProperty.java

```diff
@@ -207,8 +205,7 @@ public void finalizeValueOnRead() {
 
     @Override
     public void implicitFinalizeValue() {
-        state.disallowChanges();
-        state.finalizeOnNextGet();
+        state.disallowChangesAndFinalizeOnNextGet();
```

Comment:
So it can be reused in other contexts (I learned later that CFC doers *not* prevent further changes btw, only reads).